include/solvers/multiphysics_interface.h

```diff
@@ -540,6 +540,46 @@ class MultiphysicsInterface
     return reynolds_stress_solutions;
   }
 
+  /**
+   * @brief Request the solid objects. Used when a heat boundary condition
+   * is applied on a solid, through Nitsche immersed boundary method.
+   */
+  std::vector<std::shared_ptr<SolidBase<dim, dim>>> *
+  get_solids()
+  {
+    return solids;
```

Comment:
I would add an assertion here based on the number of solids or something like this for defensive programming.

Reply:
I tried to implement it as you suggested, but it was just never triggered. This is because `get_solids` in the `multiphysics_interface` is called only by the `heat_transfer` solver, in the method `assemble_nitsche_restriction`, which is itself is called only if `number_solids > 0`, in the method `assemble_matrix_and_rhs` (still in the `heat_transfer.cc`). I will specify it in the description of this method because it took me a while to pinpoint that! 

We do not have `set nitsche = true` anywhere (unlike for subphysics), that could be used to assert that if we use NIB, we have `number of solids` getter than 0. I think that a cleaner way to do so would be to assert that if the solver `gls_nitsche_navier_stokes` is called, a solid should be defined (and likewise, if another solver is called but `number of solids` is not 0 in the prm, a warning is thrown). Maybe by asserting it in the solvers directly? I do not know if this would have a negative impact for testing/debugging. What do you think?

Reply:
Well if this can't be called if there are no solids, then it won't happen, but I would still add an assertion here even if it never happens. Better be more defensive than not enough.
It's all right to write assertions which may never be triggered. The more we have, the better it is. Assertions have zero cost in release mode
:)